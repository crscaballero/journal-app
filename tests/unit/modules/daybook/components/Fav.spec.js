import { shallowMount } from '@vue/test-utils';

import Fab from '@/modules/daybook/components/Fab.vue';

describe('Fab.vue', () => {
    test('It must render the component correctly', () => {
        const wrapper = shallowMount(Fab);
        expect(wrapper.html()).toMatchSnapshot();
    });
    test('It must show the icon for default: fa-plus', () => {
        const wrapper = shallowMount(Fab);
        const iTag = wrapper.find('i');
        expect(iTag.classes('fa-plus')).toBeTruthy()
    });
    test('It must show the icon given a prop: fa-circle', () => {
        const wrapper = shallowMount(Fab, {
            props: {
                icon: 'fa-circle'
            }
        });
        const iTag = wrapper.find('i');
        expect(iTag.classes('fa-circle')).toBeTruthy()
    });
    test('It must emit the event on:click after a click', () => {
        const wrapper = shallowMount(Fab);
        wrapper.find('button').trigger('click');
        expect(wrapper.emitted('on:click')).toHaveLength(1);
    });
});
