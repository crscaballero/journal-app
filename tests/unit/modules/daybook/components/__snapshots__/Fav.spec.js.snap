// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab.vue It must render the component correctly 1`] = `<button type="button" class="btn btn-primary"><i class="fa fa-2x fa-plus fa-plus"></i></button>`;
